framework/src/modules/interoperability/sidechain/commands/mainchain_registration.ts

```diff
@@ -193,11 +200,44 @@ export class MainchainRegistrationCommand extends BaseInteroperabilityCommand {
 		await outboxRootSubstore.set(context, MAINCHAIN_ID_BUFFER, { root: EMPTY_HASH });
 
 		const ownChainAccountSubstore = this.stores.get(OwnChainAccountStore);
-		await ownChainAccountSubstore.set(context, utils.intToBuffer(0, 4), {
+		await ownChainAccountSubstore.set(context, EMPTY_BYTES, {
 			name: ownName,
 			id: ownChainID,
 			nonce: BigInt(0),
 		});
+		this.events
+			.get(ChainAccountUpdatedEvent)
+			.log(methodContext, MAINCHAIN_ID_BUFFER, mainchainAccount);
+
+		const encodedParams = codec.encode(registrationCCMParamsSchema, {
+			chainID: MAINCHAIN_ID_BUFFER,
+			name: MAINCHAIN_NAME,
+			messageFeeTokenID: TOKEN_ID_LSK_MAINCHAIN,
+		});
+
+		const ownChainAccount = await ownChainAccountSubstore.get(context, EMPTY_BYTES);
+
+		const ccm = {
+			nonce: ownChainAccount.nonce,
+			module: MODULE_NAME_INTEROPERABILITY,
+			crossChainCommand: CROSS_CHAIN_COMMAND_REGISTRATION,
+			sendingChainID: ownChainAccount.id,
+			receivingChainID: MAINCHAIN_ID_BUFFER,
+			fee: BigInt(0),
+			status: CCM_STATUS_OK,
+			params: encodedParams,
+		};
+		const interoperabilityStore = this.getInteroperabilityStore(context);
+		await interoperabilityStore.addToOutbox(MAINCHAIN_ID_BUFFER, ccm);
+
+		ownChainAccount.nonce += BigInt(1);
+		await ownChainAccountSubstore.set(context, EMPTY_BYTES, ownChainAccount);
+
+		const ccmID = utils.hash(codec.encode(ccmSchema, ccm));
+		this.events.get(CcmProcessedEvent).log(methodContext, ownChainAccount.id, MAINCHAIN_ID_BUFFER, {
```

Comment:
In the LIP data is `{"ccmID": ccmID, "result": CCM_SENT, "code": CODE_OK}`

```python
emitEvent(
        module = MODULE_NAME_INTEROPERABILITY,
        name = EVENT_NAME_CCM_SENT,
        data = {"ccmID": ccmID, "result": CCM_SENT, "code": CODE_OK},
        topics = [ccm.sendingChainID, ccm.receivingChainID, ccmID]
    )
```

Reply:
The type `CcmProcessedEventData` and it's usage comes from the previous [PR](https://github.com/LiskHQ/lisk-sdk/pull/7500). I think the LIP uses the same type for both `mainchain_registration.ts` and `sidechain_registration.ts`

Reply:
I'm not sure but maybe on the way the LIP update PR got updated and now I see this event data has more fields. Can you double check from your end?

https://github.com/LiskHQ/lips/blob/91aac5c1b3256ec1d4fc26593e6f5be5c3f5cdcd/proposals/lip-0043.md#execution-1

Reply:
yea, this particular change comes from a [commit](https://github.com/LiskHQ/lips/pull/148/commits/c573e3c563f65b8c9a9a302d4ddbe2892b7be9c4#diff-68930ce9364cbf6310b1c5ebd2f9137d521fa48ff3e317a345fa3f3f815c195aL339) made yesterday.

Reply:
I am assuming recent updates to the [LIP PR](https://github.com/LiskHQ/lips/pull/148) (commits from yesterday etc.) and further updates will be tackled in a new issue unless you have a different idea.

Reply:
Created issue https://github.com/LiskHQ/lisk-sdk/issues/7550